src/containers/AssetList.js

```diff
@@ -13,22 +15,42 @@ const TITLES = {
   '/assets/all': 'Assets: All',
 };
 
-const AssetList = ({ match }) => (
-  <Page>
-    <AssetListHeader title={TITLES[match.url]} />
+class AssetList extends Component {
+  componentDidMount() {
+    const { getAssets, lastAssetListUrl } = this.props;
 
-    {/* Table of currently loaded assets. */}
-    <AssetTable />
+    // Fetch an asset list if one has not already been fetched. We detect a fetch by looking at the
+    // "url" property in the assets state which is exposed as the "lastAssetListUrl" prop.
+    if(!lastAssetListUrl) {
+      getAssets({ sort: { field: 'updated_at', direction: Direction.descending } });
+    }
+  }
 
-    <div style={{textAlign: 'center'}}>
-      {/* When this component becomes visible more assets are loaded to populate the table. */}
-      <GetMoreAssets />
-    </div>
-  </Page>
-);
+  render() {
+    const { match } = this.props;
+    return (
+      <Page>
+        <AssetListHeader title={TITLES[match.url]} />
+
+        {/* Table of currently loaded assets. */}
+        <AssetTable />
+
+        <div style={{textAlign: 'center'}}>
+          {/* When this component becomes visible more assets are loaded to populate the table. */}
+          <GetMoreAssets />
```

Comment:
Is it possible for GetMoreAsset to keep doing all the asset getting? (just asking the question)

Reply:
Unsure. I'll have a look.

Reply:
The answer is "possibly" but I think it might need a separate story.

Reply:
To be a bit more explicit: I'd like to give the churn we have in play a moment to settle before refactoring this. With 5 or 6 PRs flying around, I'd like to be as compartmentalised as possible for the moment and then look back at the end of the sprint for things like this.